### PR TITLE
Reland "Add more tests to slider to avoid future breakages (#98772)"

### PR DESCRIPTION
**Pull request by goderbauer**
This reverts commit 9aed29873a3c8f4b8897224cf8291dcddd60e54a.

It's not super-clear to me why this change failed in post-submit. All checks on the original PR #98772 were green and the new images were properly triaged on the gold dashboard prior to submitting the PR.

@Piinks Is there anything else I need to do to land this cleanly?

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/98936

File: packages/flutter/test/material/value_indicating_slider_test.dart [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    await _buildValueIndicatorStaticSlider(
+      tester,
+      value: 0,
+    );
+
+    await _pressStartThumb(tester);
+
+    await expectLater(
+      find.byType(MaterialApp),
+      matchesGoldenFile('slider_start_text_scale_1_width_0'),
```
Comment (Piinks) [inline packages/flutter/test/material/value_indicating_slider_test.dart:23]: These all need to end with `.png`

**Review by Piinks — CHANGES_REQUESTED** [review]
These failed in post submit because without `.png` they all resolved to the same test name `material`, but they are all different images.

**Comment by goderbauer** [comment]
Thanks, @Piinks. Fixes the file names and filed https://github.com/flutter/flutter/issues/98943 to (hopefully) catch this with an assertion in the future.

**Comment by skia-gold** [comment]
Gold has detected about 8 new digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/98936

**Review by Piinks — APPROVED** [review]
Flutter_LGTM

**Pull request merged**